helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -237,6 +239,7 @@ private void registerMessageHandlerFactory(String type, MessageHandlerFactory fa
     }
 
     _isShuttingDown = false;
+    _isResetComplete = false;
```

Comment:
NIT: this is more like an action flag instead of state flag. Personally, I prefer flag like isCleanState or something else.